src/poppler/document.py

```diff
@@ -58,24 +59,26 @@ def __init__(self, poppler_document, data=None):
             raise ValueError("Corrupted or invalid PDF document")
 
         self._document = poppler_document
-        self._data = data
+        self._data: bytes = data
 
     @ensure_unlocked
-    def create_font_iterator(self, page=0):
+    def create_font_iterator(self, page: int = 0) -> FontIterator:
         return FontIterator(self._document.create_font_iterator(page))
 
     @ensure_unlocked
-    def create_toc(self):
+    def create_toc(self) -> Union[Toc, None]:
         t = self._document.create_toc()
         return Toc(t) if t else None
 
     @ensure_unlocked
-    def create_page(self, index):
+    def create_page(self, index: int) -> Page:
+        if not isinstance(index, int):
+            raise TypeError("index MUST be int")
```

Comment:
Why do you need that check? If index is not an int, it will raise an exception anyway...

Reply:
And would that help the user about the right type for `index`?